app/assets/javascripts/discourse/app/components/discovery/filter-tips.gjs

```diff
@@ -0,0 +1,552 @@
+      { value: "7", key: "last_week" },
+      { value: "30", key: "last_month" },
+      { value: "365", key: "last_year" },
+    ];
+
+    return dateOptions
+      .filter((option) => {
+        const description = i18n(`filter.description.days.${option.key}`);
+        return (
+          !lastTerm ||
+          option.value.includes(lastTerm) ||
+          description.toLowerCase().includes(lastTerm.toLowerCase())
+        );
+      })
+      .map((option) => ({
+        name: `${prefix}${filterName}:${prevTerms}${option.value}`,
+        description: i18n(`filter.description.${option.key}`),
+        isPlaceholderCompletion: true,
+        term: option.value,
+      }));
+  }
+
+  @action
+  setupEventListeners(element) {
+    this.inputElement = this.args.inputElement;
+
+    if (!this.inputElement) {
+      throw new Error(
+        "FilterTips requires an inputElement to be passed in the args."
+      );
+    }
+    this.element = element;
+    this.inputElement.addEventListener("focus", this.handleInputFocus);
+    this.inputElement.addEventListener("blur", this.handleInputBlur);
+    this.inputElement.addEventListener("keydown", this.handleKeyDown);
+    this.inputElement.addEventListener("input", this.handleInput);
+  }
+
+  @action
+  handleInput() {
+    this.currentInputValue = this.inputElement.value;
+    this.updateResults();
+  }
+
+  updateResults() {
+    this.selectedIndex = -1;
+
+    const words = this.currentInputValue.split(/\s+/);
+    const lastWord = words[words.length - 1];
+    const colonIndex = lastWord.indexOf(":");
+
+    if (colonIndex > 0) {
+      const prefix = this.extractPrefix(lastWord);
+      const filterName = lastWord.substring(
+        prefix.length,
+        colonIndex + prefix.length
+      );
+      const valueText = lastWord.substring(colonIndex + 1);
+
+      const tip = this.args.tips.find((t) => {
+        const tipFilterName = t.name.replace(/^[-=]/, "").split(":")[0];
+        return tipFilterName === filterName && t.type;
+      });
+
+      if (tip?.type) {
+        this.activeFilter = filterName;
+        this.handlePlaceholderSearch(filterName, valueText, tip, prefix);
+      } else {
+        this.activeFilter = null;
+        this.searchResults = [];
+      }
+    } else {
+      this.activeFilter = null;
+      this.searchResults = [];
+    }
+  }
+
+  @action
+  handleInputFocus() {
+    this.currentInputValue = this.inputElement.value;
+    this.showTips = true;
+    this.selectedIndex = -1;
+  }
+
+  @action
+  handleInputBlur() {
+    if (this.handleBlurTimer) {
+      cancel(this.handleBlurTimer);
+    }
+    // delay this cause we need to handle click events on tips
+    this.handleBlurTimer = later(() => {
+      this.hideTipsIfNeeded();
+    }, 200);
+  }
+
+  hideTipsIfNeeded() {
+    this.handleBlurTimer = null;
+    if (document.activeElement !== this.inputElement && this.showTips) {
+      this.hideTips();
+    }
+  }
+
+  @action
+  handleKeyDown(event) {
```

Comment:
Really would be good to have an ember modifier or something for behaviour like this eventually, this is at least the third time it's come up (last time was on admin search which has to handle similar things)

Reply:
agree, would be nice, we should raise on dev.